### PR TITLE
Add broker endpoint to handle backup status query

### LINKED ISSUE #9807: Broker can receive backup status query
Depends on #9806 

Broker can receive a request to query the status of a backup of a partition from the gateway and send the response back.
- Define request format
- Define the response format
- Implement and endpoint in Broker to handle the request. 
- The handler will use the backup manager (#9806) to get the status info

### PR DESCRIPTION
**Pull request by deepthidevaki**
## Description

- BackupApiRequestHandler can handle backup status query
- Added a new response type for BackupStatus.

## Related issues

closes #9807 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [x] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

**Review by lenaschoenburg — APPROVED** [review]
Looks good to me :+1: 
Please consider one suggestion before merging.

File: protocol-impl/src/main/java/io/camunda/zeebe/protocol/impl/encoding/BackupStatusResponse.java [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+    brokerId = bodyDecoder.brokerId();
+    checkpointPosition = bodyDecoder.checkpointPosition();
+    numberOfPartitions = bodyDecoder.numberOfPartitions();
+    snapshotId = bodyDecoder.snapshotId();
+    failureReason = bodyDecoder.failureReason();
+  }
+
+  @Override
+  public int getLength() {
+    return 0;
```
[the remark this answers is not in the document]
Comment (deepthidevaki) [reply protocol-impl/src/main/java/io/camunda/zeebe/protocol/impl/encoding/BackupStatusResponse.java:126]: :+1: I missed this. Thanks.

[the remark this answers is not in the document]
Comment (oleschoenburg) [reply protocol-impl/src/main/java/io/camunda/zeebe/protocol/impl/encoding/BackupStatusResponse.java]: Oh, and I missed that just header length + body block length isn't even enough :facepalm:

[the remark this answers is not in the document]
Comment (deepthidevaki) [reply protocol-impl/src/main/java/io/camunda/zeebe/protocol/impl/encoding/BackupStatusResponse.java]: It should also include the length of variable length fields snapshotId and failure reason. @oleschoenburg Could you please check the latest commit, before I merge?

File: protocol-impl/src/main/java/io/camunda/zeebe/protocol/impl/encoding/BackupStatusResponse.java [hunk 2]
```diff
@@ -117,13 +126,35 @@ public void wrap(final DirectBuffer buffer, final int offset, final int length)
...
+    }
   }
 
   @Override
   public int getLength() {
-    return 0;
+    return headerEncoder.encodedLength()
+        + bodyEncoder.sbeBlockLength()
+        + BackupStatusResponseEncoder.snapshotIdHeaderLength()
+        + encodedSnapshotId.length
```
Comment (oleschoenburg) [inline protocol-impl/src/main/java/io/camunda/zeebe/protocol/impl/encoding/BackupStatusResponse.java:155]: :question: Can't we just use bodyEncoder.encodedLength()? That way we also wouldn't need to maintain two variables (byte array + decoded string) each for snapshotId and failureReason.

Comment (deepthidevaki) [reply protocol-impl/src/main/java/io/camunda/zeebe/protocol/impl/encoding/BackupStatusResponse.java:155]: No. `write` may not have been called before `getLength`. bodyEncoder.encodedLength() is only correct after `write` is called.

**Comment by deepthidevaki** [comment]
bors merge

**Pull request merged**